12-java-tcp-programming/03-repl-client-server/TcpReplServerExample.java

```diff
@@ -0,0 +1,105 @@
+import java.io.*;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.nio.charset.StandardCharsets;
+
+class TcpReplServerExample {
+
+  private static final int PORT = 1234;
+  public static String END_OF_LINE = "\n";
+
+  // Constants for messages
+  public enum ClientCommand {
+    HELLO,
+  }
+
+  public enum ServerCommand {
+    HI,
+    INVALID
+  }
+
+  public static void main(String[] args) {
+    try (ServerSocket serverSocket = new ServerSocket(PORT)) {
+      System.out.println("[Server] Listening on port " + PORT);
+
+      while (!serverSocket.isClosed()) {
```

Comment:
Is it really necessary to loop over `serverSocket.isClosed()` ?

Reply:
While totally valid, IntelliJ doesn't like the condition `while (true)` (it displays a warning) and it made me thought of a better way to manage/demonstrate it.

If the application, for some reason, would close its server socket inside the loop (like a `SHUTDOWN` command from one of the client), this would still work as intended thanks to the `!serverSocket.isClosed()` instead of the `while (true)`.

It is a more elegant way but doesn't change much in this use-case, I agree.